### ISSUE TITLE
Unable to hear pipeline outputs or see transcriptions in my audio pipeline

### ISSUE THREAD

**Issue by gaceladri** [pipecat-ai/pipecat#244]
I am experiencing issues with my audio processing pipeline. When I comment out transport.output() in the pipeline, I can see the interaction logs without hearing the audio from ElevenLabs. However, if I uncomment that line, I neither hear the audio output nor see the transcription logs from Deepgram.

Here is the relevant part of my code:
```
import asyncio
import os
import sys

import aiohttp
from loguru import logger
from pipecat.frames.frames import Frame, TranscriptionFrame
from pipecat.pipeline.pipeline import Pipeline
from pipecat.pipeline.runner import PipelineRunner
from pipecat.pipeline.task import PipelineParams, PipelineTask
from pipecat.processors.aggregators.llm_response import (
    LLMAssistantResponseAggregator,
    LLMUserResponseAggregator,
)
from pipecat.processors.frame_processor import FrameDirection, FrameProcessor
from pipecat.services.deepgram import DeepgramSTTService
from pipecat.services.elevenlabs import ElevenLabsTTSService
from pipecat.services.openai import OpenAILLMService
from pipecat.transports.base_transport import TransportParams
from pipecat.transports.local.audio import LocalAudioTransport
from pipecat.vad.silero import SileroVADAnalyzer
from pipecat.vad.vad_analyzer import VADParams

logger.remove(0)
logger.add(sys.stderr, level="DEBUG")


class TranscriptionLogger(FrameProcessor):
    async def process_frame(self, frame: Frame, direction: FrameDirection):
        await super().process_frame(frame, direction)

        if isinstance(frame, TranscriptionFrame):
            print(f"Transcription: {frame.text}")


async def main():
    async with aiohttp.ClientSession() as session:
        transport = LocalAudioTransport(
            TransportParams(
                audio_in_enabled=True,
                audio_out_enabled=True,
                transcription_enabled=True,
                vad_enabled=True,
                vad_analyzer=SileroVADAnalyzer(params=VADParams(min_volume=0.6)),
                vad_audio_passthrough=True,
            )
        )

        stt = DeepgramSTTService(os.environ["DEEPGRAM_API_KEY"])

        llm = OpenAILLMService(
            api_key=os.environ["OPENAI_API_KEY"],
            model="gpt-3.5-turbo-0125",
        )

        messages = [
            {
                "role": "system",
                "content": "Say hello.",
            },
        ]

        tma_in = LLMUserResponseAggregator(messages)
        tma_out = LLMAssistantResponseAggregator(messages)

        tts = ElevenLabsTTSService(
            aiohttp_session=session,
            api_key=os.environ["ELEVENLABS_API_KEY"],
            voice_id=os.environ["ELEVENLABS_VOICE_ID"],
        )

        pipeline = Pipeline(
            [
                transport.input(),
                stt,
                tma_in,
                llm,
                tts,
                transport.output(),
                tma_out,
            ]
        )

        task = PipelineTask(pipeline, PipelineParams(allow_interruptions=True))

        runner = PipelineRunner()

        await runner.run(task)


if __name__ == "__main__":
    asyncio.run(main())

```

Additionally, when I add an introductory message from the LLM, I can hear the initial message, but the interaction with the LLM stops. Here is the code snippet for that:
```
    async def say_something():
            messages.append(
                {
                    "role": "system",
                    "content": "Please briefly introduce yourself to the user.",
                }
            )
            await task.queue_frames([LLMMessagesFrame(messages)])

        # await runner.run(task)
        await asyncio.gather(runner.run(task), say_something())
```

I am using a Mac and pipecat-ai==0.0.31. Do you see any obvious issues in my code, or could the problem be elsewhere?

**Comment by aconchillo** [comment]
If you can hear initial messages from LLM that's good. It means ElevenLabs and the output are working well.

Do you see the transcription messages from Deepgram? Also, can you see `User started speaking` and `User stopped speaking` messages in the logs?>

**Comment by gaceladri** [comment]
Thank you for your response. When I comment out the `transport.output()` method, I can see the `User started speaking` and `User stopped speaking` logs, and everything functions as expected. However, when I include the `transport.output()` method, I only see the initial message from ElevenLabs, and after that, I don't see any logs or interactions. It's as if the pipeline stops processing any further interactions.

To summarize:

- With `transport.output()` commented out: All logs and interactions work perfectly.
- With t`ransport.output()` included: I hear the initial ElevenLabs message, but no subsequent logs or interactions are processed.
 
Any guidance on what might be causing this behavior would be greatly appreciated.

**Comment by aconchillo** [comment]
This sounds like the `DeepgramSTTService` is not working properly... I will try and let you know.

**Comment by aconchillo** [comment]
I just tried your exact code and it works for me. I tried without the `say_something()` in which case you just need to say something and the bot will start answering.

**Comment by gaceladri** [comment]
There must be something wrong with my Mac then. Closing this as it is not reproducible. Thank you for checking it.

**Comment by dk-crazydiv** [comment]
```
2024-08-13 21:54:41.285 | INFO     | pipecat.transports.services.daily:_start_transcription:288 - Enabling transcription with settings language='en' tier='nova' model='2-conversationalai' profanity_filter=True redact=False endpointing=True punctuate=True includeRawResponse=True extra={'interim_results': True}
2024-08-13 21:54:42.685 | DEBUG    | pipecat.transports.services.daily:on_transcription_started:495 - Transcription started: {'tier': 'nova', 'language': 'en', 'model': '2-conversationalai', 'transcriptId': 'a8630b4c-3757-4480-b1c8-e58ac3a153b4', 'startedBy': '3ecfa220-f87a-4925-8cdf-be704b2c7624'}
2024-08-13 21:54:43.578 | DEBUG    | pipecat.services.deepgram:start:136 - DeepgramSTTService#0: Connected to Deepgram
2024-08-13 21:54:43.579 | DEBUG    | pipecat.services.openai:_stream_chat_completions:103 - Generating chat: [{"content": "You are a helpful LLM in a WebRTC call. Respond creatively and helpfully to user inputs in a conversational manner.", "role": "system", "name": "system"}, {"content": "Please welcome aa.", "role": "system", "name": "system"}]
2024-08-13 21:54:44.383 | DEBUG    | pipecat.services.elevenlabs:run_tts:42 - Generating TTS: [Thank you!]
2024-08-13 21:54:45.258 | DEBUG    | pipecat.services.elevenlabs:run_tts:42 - Generating TTS: [Hi there!]
2024-08-13 21:54:46.157 | DEBUG    | pipecat.services.elevenlabs:run_tts:42 - Generating TTS: [How can I assist you today?]
2024-08-13 21:54:46.946 | DEBUG    | pipecat.services.elevenlabs:run_tts:42 - Generating TTS: [😊]


2024-08-13 21:54:59.476 | DEBUG    | pipecat.transports.base_input:_handle_interruptions:136 - User started speaking
2024-08-13 21:55:01.136 | DEBUG    | pipecat.transports.base_input:_handle_interruptions:139 - User stopped speaking
```


Facing a similar issue here. I am on M3 mac. building the project in conda. I've been trying to get basic example to run with `daily` and I only am able to hear the first message (as logged above). After that I do get "started" and stopped speaking messages but both with Deepgram and Local whisper, I am not getting any output. Tried with public rooms, private rooms both.

**Comment by dk-crazydiv** [comment]
Update:

```
params=DailyParams(audio_in_enabled=True, audio_out_enabled=True, vad_enabled=True,
                vad_analyzer=SileroVADAnalyzer(),
                transcription_enabled=True,
                vad_audio_passthrough=True)
```
I was missing `vad_audio_passthrough=True`. Got it to run both on local and daily after I added it.
